x/clp/keeper/provider_distribution.go

```diff
@@ -132,14 +132,16 @@ func (k Keeper) CollectProviderDistributions(ctx sdk.Context, pools []*types.Poo
 	lpMap := make(LpRowanMap, 0)
 	lpPoolMap := make(LpPoolMap, 0)
 
+	partitions, err := k.GetAllLiquidityProvidersPartitions(ctx)
+	if err != nil {
+		fireLPPGetLPsErrorEvent(ctx, err)
+	}
+
 	for _, pool := range pools {
-		lps, err := k.GetAllLiquidityProvidersForAsset(ctx, *pool.ExternalAsset)
-		if err != nil {
-			//k.Logger(ctx).Error(fmt.Sprintf("Getting liquidity providers for asset %s error %s", pool.ExternalAsset.Symbol, err.Error()))
-			fireLPPGetLPsErrorEvent(ctx, *pool.ExternalAsset, err)
+		lps, exists := partitions[*pool.ExternalAsset]
+		if !exists { // TODO: fire event
```

Comment:
@sifag what type of event should be fired here, and when do you we expect this to be done?

Reply:
Up to product to decide if and what kind of event they want. I just thought it might be a good idea